### PR TITLE
feat: Group API in core and extensions

### PR DESCRIPTION
**Pull request by thomaspoignant**
## This PR
- Group the API into 2 groups `OFREP core` and `OFREP extensions`.

### CODE REVIEW

File: service/openapi.yaml [hunk 1]
```diff
@@ -14,45 +14,13 @@ security:
...
-          description: Forbidden - You are not authorized to access the API
-        '500':
-          description: Internal server error
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/generalErrorResponse'
   /ofrep/v1/evaluate/flags/{key}:
     post:
+      tags: [OFREP core]
```
[the remark this answers is not in the document]
Comment (thomaspoignant) [reply service/openapi.yaml:19]: They are needed
Screenshot 2024-06-07 at 18 39 52

[the remark this answers is not in the document]
Comment (thomaspoignant) [reply service/openapi.yaml:19]: The tag object that you linked is for the schema level.

Here we are at operation level and **Tag object** is not supported.
See https://spec.openapis.org/oas/v3.1.0#operation-object

Screenshot 2024-06-07 at 18 45 40

**Review by beeme1mr — APPROVED** [review]

File: service/openapi.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (grimly) [reply service/openapi.yaml:19]: I know, that's what I was suggesting you add. But github dont let you add review on unmodified lines

[the remark this answers is not in the document]
Comment (thomaspoignant) [reply service/openapi.yaml]: Oh, I didn't know about this one.
I have added it.

File: service/openapi.yaml [hunk 2]
```diff
@@ -13,46 +13,29 @@ info:
...
+      
+      *This is the minimum set of APIs required for a flag management system to be OFREP compatible.*
+      
+  - name: OFREP extensions
+    description: |
+      Extension APIs to provide full support for OFREP.  
+      **Optional**  
+
+      *These APIs are utilized by the providers to enhance the OFREP experience.*
+
```
Comment (Kavindu-Dodan) [inline service/openapi.yaml:30]:
```suggestion
tags:
  - name: OFREP Core
    description: |
      **Required**: Core APIs to implement to support OFREP.  
      *This is the minimum set of APIs required for a flag management system to be OFREP compatible.*
  - name: OFREP Extensions
    description: |
      **Optional**: Extension APIs to provide full support for OFREP.  
      *These APIs are utilized by the providers to enhance the OFREP experience.*
      
```

I know naming is hard, but I think we could use OFREP Core & OFREP Extensions capitalized terms. And the formatting looks good ,

image

**Review by Kavindu-Dodan — APPROVED** [review]
I have this suggestion [1], otherwise this is a good improvement :) 

[1] - https://github.com/open-feature/protocol/pull/23/files#r1631670185

**Pull request merged**